src/firefly/java/edu/caltech/ipac/firefly/core/FileAnalysisReport.java

```diff
@@ -182,7 +182,8 @@ void makeBrief() {
     public static class Part {
         private final Type type;
         private UIRender uiRender=UIRender.NotSpecified;
-        private UIEntry uiEntry= UIEntry.UseAll;;
+        private UIEntry uiEntry= UIEntry.UseAll;
+        private String url;
```

Comment:
I don't quite see the problem. The `Part` is generic enough the there could be other cases with URL. There are table and chart params as well. It could be a big inheritance but that would be a mess. It is there to turn into a javascript which will pick only that stuff that is used. I guess we could use a `Map`, that might be the best.

Reply:
Yes, I was thinking a Map.  By adding a variable, you need to add setter/getter, then modify the serializer(toJsonSting) just so it will turn back to a JSON map.  The setter/getter here doesn't help on the client side, and it's not very useful on the server either.  A set of constants for the keys would have been the same.

Reply:
Should I leave this as is or make changes? added a comment for now saying 'url' is only used for UWS.